src/commands/ask.ts

```diff
@@ -0,0 +1,118 @@
+import { ApplicationCommandOptionType, CommandInteraction, InteractionResponse, Message } from 'discord.js'
+import {
+  Discord,
+  SimpleCommand,
+  SimpleCommandMessage,
+  SimpleCommandOption,
+  SimpleCommandOptionType,
+  Slash,
+  SlashOption,
+} from 'discordx'
+
+const COMMAND_NAME = 'ask'
+const COMMAND_DESC = 'Ask a question to Wolfram Alpha'
+
+const COOLDOWN_MILLISECONDS = 3 * 60 * 1000
+
+@Discord()
+class Ask {
+  private apiToken: string
+  private lastUsage: number
+
+  constructor() {
+    this.apiToken = process.env.WOLFRAM_APP_ID ?? ''
+    this.lastUsage = 0
+
+    if (this.apiToken == '') {
+      throw Error('WOLFRAM_APP_ID needs to be set')
+    }
+  }
+
+  @SimpleCommand({ name: COMMAND_NAME, description: COMMAND_DESC })
+  async simple(
+    @SimpleCommandOption({ name: 'question', type: SimpleCommandOptionType.String })
+    question: string | undefined,
+    command: SimpleCommandMessage
+  ) {
+    if (!question) {
+      return await command.message.reply({ content: 'Usage: >ask <question>', allowedMentions: { repliedUser: false } })
+    }
+
+    if (this.isOnCooldown()) {
+      return
+    }
+
+    try {
+      const answer = await this.fetchAnswer(question)
+      return command.message.reply({ content: answer, allowedMentions: { repliedUser: false } })
```

Comment:
Does this text need to be escaped of formatting markdown etc.? Will an answer with things like 3  * 4 * 5 end up as "3 4 5"?
Maliciously, I wonder if there is a way to get this to make replies that ping users or do other funny business.

Maybe this is already dealt with by discord?

Reply:
Discord visually converts the text to markdown but the question sent to the API is what you'd expect (`3 *4* 5 = 60`), so that's not a problem.
I did change it so the API answer and the question that gets referenced in the slash command get escaped though.

Reply:
Yeah I was thinking mostly if the reply text from Wolfram would get mangled by discord.